node/cmd/guardiand/adminclient.go

```diff
@@ -168,6 +179,29 @@ func getPublicRPCServiceClient(ctx context.Context, addr string) (*grpc.ClientCo
 	return conn, c, err
 }
 
+func runSignWormchainValidatorAddress(cmd *cobra.Command, args []string) {
+	guardianKeyPath := args[0]
+	wormchainAddress := args[1]
+	if !strings.HasPrefix(wormchainAddress, "wormhole") || strings.HasPrefix(wormchainAddress, "wormholeval") {
+		log.Fatalf("must provide a bech32 address that has 'wormhole' prefix")
+	}
+	gk, err := loadGuardianKey(guardianKeyPath)
+	if err != nil {
+		log.Fatalf("failed to load guardian key: %v", err)
+	}
+	addr, err := cosmostypes.GetFromBech32(wormchainAddress, "wormhole")
+	if err != nil {
+		log.Fatalf("failed to decode wormchain address: %v", err)
+	}
+	// Hash and sign address
+	addrHash := crypto.Keccak256Hash(append(sdk.SignedWormchainAddressPrefix, addr...))
```

Comment:
So append is a bit weird because it can actually extend or overwrite the source slice if it has sufficient capacity.  See https://go.dev/play/p/yDdUCBV9nnW.

I think in this case we *should* be fine because hopefully the runtime doesn't allocate additional capacity for `SignedWormchainAddressPrefix` but if we wanted to be really paranoid, we would copy the slice first:

```go
msg := make([]byte, 0, len(sdk.SignedWormchainAddressPrefix) + len(addr))

count := copy(msg, sdk.SignedWormchainAddressPrefix)
copy(msg[count...], addr)
addrHash := crypto.Keccak256Hash(msg)
```

Same comment for the other couple of places where we append directly.

Reply:
Good catch!  Will start using this copy pattern now.  I resolved using the variadic args straight to `Keccak256Hash(...)` for this one.